lib/Sema/CSGen.cpp

```diff
@@ -3473,6 +3474,21 @@ bool swift::isExtensionApplied(DeclContext &DC, Type BaseTy,
   ConstraintSystem CS(*TC, &DC, Options);
   auto Loc = CS.getConstraintLocator(nullptr);
 
+  // Open up the type parameters, if needed.
+  auto allowFreeTypeVariables = FreeTypeVariableBinding::Disallow;
+  if (openTypeParameters && BaseTy->hasTypeParameter()) {
+    // Open up the type parameters.
+    constraints::OpenedTypeMap replacements;
+    CS.openGeneric(&DC, DC.getModuleScopeContext(),
+                   DC.getGenericSignatureOfContext(),
+                   /*skipProtocolSelfConstraint=*/false,
+                   Loc, replacements);
+    BaseTy = CS.openType(BaseTy, replacements);
+
+    // Allow free type variables.
+    allowFreeTypeVariables = FreeTypeVariableBinding::Allow;
```

Comment:
Looks like we aren't using `allowFreeTypeVariables` anywhere, but it should be passed to `solveSingle`

Reply:
On success, need to map remaining constraints back to Requirements for the caller